src/types/redis_string.cc

```diff
@@ -371,28 +371,33 @@ rocksdb::Status String::MSet(const std::vector<StringPair> &pairs, uint64_t ttl,
     expire = now + ttl;
   }
 
-  // Data race, key string maybe overwrite by other key while didn't lock the key here,
+  // Data race, key string maybe overwrite by other key while didn't lock the keys here,
   // to improve the set performance
   std::string ns_key;
```

Comment:
Can we move `ns_key` in the loop?

```c++
      std::string ns_key
      AppendNamespacePrefix(pair.key, &ns_key);
      lock_keys.emplace_back(std::move(ns_key));
    }

Reply:
In the code below, we also use `ns_key`
```
  auto batch = storage_->GetWriteBatchBase();
  WriteBatchLogData log_data(kRedisString);
  batch->PutLogData(log_data.Encode());
  for (const auto &pair : pairs) {
    std::string bytes;
    Metadata metadata(kRedisString, false);
    metadata.expire = expire;
    metadata.Encode(&bytes);
    bytes.append(pair.value.data(), pair.value.size());
    AppendNamespacePrefix(pair.key, &ns_key);
    batch->Put(metadata_cf_handle_, ns_key, bytes);
  }
```

Reply:
Could it be `move` here?

`ns_key` is built and used instantly, seems that it would not keep it

Reply:
can you describe it a bit more or insert a suggestion? 
i am not very familiar with this. (not a pure C++ coder)

Reply:
```c++
    for (const StringPair &pair : pairs) {
      std::string ns_key;
      AppendNamespacePrefix(pair.key, &ns_key);
      lock_keys.emplace_back(std::move(ns_key));
    }
```

And 

```c++
  for (const auto &pair : pairs) {
    std::string bytes;
    Metadata metadata(kRedisString, false);
    metadata.expire = expire;
    metadata.Encode(&bytes);
    bytes.append(pair.value.data(), pair.value.size());
    std::string ns_key;
    std::optional<MultiLockGuard> guard;
  if (lock) {
    std::vector<std::string> lock_keys;
    lock_keys.reserve(pairs.size());
    for (const StringPair &pair : pairs) {
      AppendNamespacePrefix(pair.key, &ns_key);
      lock_keys.emplace_back(ns_key);
    }
    guard.emplace(storage_->GetLockManager(), lock_keys);
  }
```

I suggest this one. But it's ok to keep the current implemention

Reply:
i see in the second code block, the lock part is inside the for loop? is that a copy-paste error?

Reply:
Yeah just an example..

Reply:
ohh, ok i got your point (finally!), thanks for the review.